cloudinit/features.py

```diff
@@ -87,6 +87,31 @@
 to write /etc/apt/sources.list directly.
 """
 
+DEPRECATION_INFO_BOUNDARY = "devel"
+"""
+DEPRECATION_INFO_BOUNDARY is used by distros to configure at which version to
+start logging deprecations at a level higher than INFO.
+
+The default value "devel" tells cloud-init to log all deprecations higher
+than INFO. This value may be overriden by downstreams in order to maintain
+stable behavior across releases.
+
+Jsonschema key deprecations and inline logger deprecations include a
+deprecated_version key. When the variable below is set to a version,
+cloud-init will use that version as a demarcation point. Deprecations which
+are added after this version will be logged as at an INFO level. Deprecations
+which predate this version will be logged at the higher DEPRECATED level.
+Downstreams that want stable log behavior may set the variable below to the
+first version released in their stable distro. By doing this, they can expect
+that newly added deprecations will be logged at INFO level.
+
+format:
+
+<value> :: = <default> | <version>
+<default> ::= "devel"
+<version> ::= <major> "." <minor> "." <patch>
```

Comment:
> Now that we are accepting a slightly more complex schema than booleans in features.py, I bet if we get more complex values for other 'features', some sort of schema validation of the configured values could be helpful to avoid the potential of downstreams. It's definitely not worth the investment on this PR, but a discussion topic we could bring up if we get more complex configurable feature values.

Eh, maybe. Personally I think that trying to prevent errors introduced by downstreams or third-parties in upstream code is a lot lower priority than [validating user-introduced errors and documenting the possible cloud.cfg values](https://github.com/canonical/cloud-init/issues/5410).